server/api/middleware.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"net/http"
+
+	"github.com/pingcap/pd/server"
+	"github.com/unrolled/render"
+)
+
+type clusterMiddleware struct {
+	s  *server.Server
+	rd *render.Render
+}
+
+func newClusterMiddleware(s *server.Server) clusterMiddleware {
+	return clusterMiddleware{
+		s:  s,
+		rd: render.New(render.Options{IndentJSON: true}),
+	}
+}
+
+func (m clusterMiddleware) Middleware(h http.Handler) http.Handler {
```

Comment:
`Middleware` is fuzzy. Can we use more accurate name? such as `CheckBoot`?

Reply:
I'm using the style of the mux example https://github.com/gorilla/mux#middleware

Reply:
Get